bond_app/authentication.py

```diff
@@ -74,17 +77,21 @@ def require_user_info(self, request_state, token_info_fn=_token_info):
 
         token = auth_header_parts[1]
 
-        user_info = self.cache_api.get(key='access_token:' + token)
-        if user_info is None:
-            user_info = self._fetch_user_info(token, token_info_fn)
+        google_user_info = self.cache_api.get(key='access_token:' + token)
+        if google_user_info is None:
+            google_user_info = self._fetch_user_info(token, token_info_fn)
             # cache for 10 minutes or until token expires
-            expires_in = min([user_info.expires_in, self.config.max_token_life])
+            expires_in = min([google_user_info.expires_in, self.config.max_token_life])
             logging.debug("caching token %s for %s seconds", token, expires_in)
-            self.cache_api.add(key='access_token:' + token, value=user_info, expires_in=expires_in)
+            self.cache_api.add(key='access_token:' + token, value=google_user_info, expires_in=expires_in)
         else:
             logging.debug("auth token cache hit for token %s", token)
 
-        return user_info
+        sam_user_info = self.sam_api.user_info(google_user_info)
```

Comment:
from a code organization perspective I would prefer that the caching code be here and leave the sam_api as just the raw sam api calls. sam_api should have limited logic, it should just do the sam things when asked (in other words, when I call sam_api I want to actually call sam). Here is a higher level bit of code that has more business logic and can be more sophisticated.